cpp/src/neighbors/batch_ann/batch_ann.cuh

```diff
@@ -0,0 +1,476 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include "../detail/ann_utils.cuh"
+#include "cuvs/neighbors/common.hpp"
+#include "cuvs/neighbors/ivf_pq.hpp"
+#include <cstddef>
+#include <cuvs/neighbors/batch_ann.hpp>
+#include <cuvs/neighbors/brute_force.hpp>
+#include <cuvs/neighbors/refine.hpp>
+#include <raft/core/device_mdarray.hpp>
+#include <raft/core/host_mdarray.hpp>
+#include <raft/core/host_mdspan.hpp>
+#include <raft/core/managed_mdarray.hpp>
+#include <raft/core/resource/nccl_clique.hpp>
+#include <rmm/mr/device/statistics_resource_adaptor.hpp>
```

Comment:
This is for debugging so we probably don't want this in the final code. We probably don't want `nccl_clique` in there either. Please cleanu p the includes in this file to only include what's actually being used.

Reply:
the `nccl_clique` is for figuring out how many visible CUDA devices there are when running the program. Is there a better way to figure that out? (to decide whether we want to go multi-gpu or single gpu)?